### PR TITLE
feat: updating last contributed column for insights table

### LINKED ISSUE #2110: Move last repo contributed to under last contributed column for contributors
## Current Issue
Currently, under the name and profile picture of the contributor, we have the name of the last repo they contributed to. 

rachel nabors, under is aws-amplify/docs

## Solution
Let's move the repo name under the time period in the last contributed column. 
image

That would require removing the second paragraph under the div with the user's name. It will look something like this: `<p class="hidden text-sm font-normal md:inline-flex text-light-slate-9">open-sauced/intro</p>`.

Next, add that paragraph under the last contributed date.

### PR DESCRIPTION
**Pull request by jdwilkin4**
## Description

This PR is responsible for moving the last contributed repo from under the contributor name to underneath the last contributed column in the insights table. 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #2110 

### Mobile(Before)
Screenshot 2023-11-10 at 11 33 47 PM

### Mobile(After)
Screenshot 2023-11-10 at 11 34 02 PM

### Desktop(Before)

Screenshot 2023-11-10 at 11 34 37 PM

### Desktop(After)

Screenshot 2023-11-10 at 11 34 48 PM

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

## Storybook 
Since I made some style changes I also created stories for the new style changes. 

### DevProfile
Screenshot 2023-11-10 at 11 58 53 PM

### Contributor list table row
Screenshot 2023-11-11 at 12 23 19 AM

### CODE REVIEW

File: components/molecules/ContributorListTableRow/contributor-list-table-row.tsx [hunk 1]
```diff
@@ -179,7 +177,12 @@ const ContributorListTableRow = ({
 
         {/* Column: Last Contribution */}
         <div className={clsx("flex-1 lg:max-w-[130px]  flex text-light-slate-11 justify-center ")}>
-          <div className="flex">{<p>{contributor.author_login ? getLastContributionDate(mergedPrs) : "-"}</p>}</div>
+          <div className="flex flex-col">
+            <p>{contributor.author_login ? getLastContributionDate(mergedPrs) : "-"}</p>{" "}
+            <p className="hidden text-sm font-normal md:inline-flex text-light-slate-9">
+              {truncateString(user?.company || getLastContributedRepo(data), 20)}
```
Comment (OgDev-01) [inline components/molecules/ContributorListTableRow/contributor-list-table-row.tsx:183]: We are incrementally moving out of this approach in truncating string, you should look at https://github.com/open-sauced/app/issues/1763 for the new approach 👍

**Review by OgDev-01 — APPROVED** [review]
Aside the comment i left, this all looks good 👏👏👏. Thanks for taking this on @jdwilkin4

File: components/molecules/ContributorListTableRow/contributor-list-table-row.tsx [hunk 2]
```diff
@@ -89,14 +89,18 @@ const ContributorListTableRow = ({
...
-              hasBorder={!contributor.author_login}
-            />
+            <DevProfile username={login} hasBorder={!contributor.author_login} />
           </div>
           <div className="w-[34%] text-normal text-light-slate-11  h-full">
-            <div className="flex gap-x-3">{<p>{getLastContributionDate(mergedPrs)}</p>}</div>
+            <div className="flex flex-col gap-x-3">
+              <p>{getLastContributionDate(mergedPrs)}</p>{" "}
+              <p
+                className="text-sm font-normal truncate whitespace-nowrap overflow-hidden overflow-ellipsis text-light-slate-9 md:hidden"
```
Comment (nickytonline) [inline components/molecules/ContributorListTableRow/contributor-list-table-row.tsx:98]: This works, but the columns are getting too large in some cases due to a long org/project name and the ellipsis isn't kicking in, e.g.

CleanShot 2023-11-13 at 15 15 03

A max width would help with this.

I think longer term we need to revisit this grid, but for now, that should help with the misalignment.

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: components/molecules/ContributorListTableRow/contributor-list-table-row.tsx [hunk 3]
```diff
@@ -89,14 +89,18 @@ const ContributorListTableRow = ({
...
-              hasBorder={!contributor.author_login}
-            />
+            <DevProfile username={login} hasBorder={!contributor.author_login} />
           </div>
           <div className="w-[34%] text-normal text-light-slate-11  h-full">
-            <div className="flex gap-x-3">{<p>{getLastContributionDate(mergedPrs)}</p>}</div>
+            <div className="flex flex-col gap-x-3">
+              <p>{getLastContributionDate(mergedPrs)}</p>{" "}
+              <p
+                className="text-sm font-normal truncate whitespace-nowrap overflow-hidden overflow-ellipsis text-light-slate-9 md:hidden lg:max-w-[130px]"
```
Comment (NsdHSO) [inline components/molecules/ContributorListTableRow/contributor-list-table-row.tsx:98]: instead `130px` why not `8.12rem`?

File: components/molecules/ContributorListTableRow/contributor-list-table-row.tsx [hunk 4]
```diff
@@ -89,14 +89,18 @@ const ContributorListTableRow = ({
...
-              hasBorder={!contributor.author_login}
-            />
+            <DevProfile username={login} hasBorder={!contributor.author_login} />
           </div>
           <div className="w-[34%] text-normal text-light-slate-11  h-full">
-            <div className="flex gap-x-3">{<p>{getLastContributionDate(mergedPrs)}</p>}</div>
+            <div className="flex flex-col gap-x-3">
+              <p>{getLastContributionDate(mergedPrs)}</p>{" "}
+              <p
+                className="text-sm font-normal truncate whitespace-nowrap overflow-hidden overflow-ellipsis text-light-slate-9 md:hidden lg:max-w-[8.12rem]"
```
Comment (OgDev-01) [inline components/molecules/ContributorListTableRow/contributor-list-table-row.tsx:98]: truncate in tailwind is a combination of 
- overflow-hidden
- text-overflow-ellipsis
- whitespace-nowrap

all you need to achieve the truncate here is the `truncate` class

```suggestion
                className="text-sm font-normal truncate text-light-slate-9 md:hidden lg:max-w-[8.12rem]"
```

**Review by OgDev-01 — CHANGES_REQUESTED** [review]

**Review by OgDev-01 — APPROVED** [review]
All good now 🍕🍕

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks @jdwilkin4!

**Pull request merged**